sdk/src/invariant.ts

```diff
@@ -53,72 +53,87 @@ export class Invariant {
   private constructor(
     api: ApiPromise,
     network: Network,
+    abi: any,
+    address: string,
     storageDepositLimit: number | null = null,
     refTime: number = DEFAULT_REF_TIME,
-    proofSize: number = DEFAULT_PROOF_SIZE,
-    abi: any,
-    deploymentAddress: string
+    proofSize: number = DEFAULT_PROOF_SIZE
   ) {
     this.api = api
+    this.waitForFinalization = network !== Network.Local
+    this.contract = new ContractPromise(this.api, abi, address)
     this.gasLimit = api.registry.createType('WeightV2', {
       refTime,
       proofSize
     }) as WeightV2
     this.storageDepositLimit = storageDepositLimit
-    this.waitForFinalization = network != Network.Local
-    this.contract = new ContractPromise(this.api, abi, deploymentAddress)
     this.abi = new Abi(abi)
   }
 
   static async getContract(
     api: ApiPromise,
-    account: IKeyringPair,
-    storageDepositLimit: number | null = null,
-    refTime: number = DEFAULT_REF_TIME,
-    proofSize: number = DEFAULT_PROOF_SIZE,
-    initFee: Percentage,
-    network: Network
+    network: Network,
+    account?: IKeyringPair,
```

Comment:
Name it `deployer: IKeyringPair`. From the interface perspective, you don't know what type of `account` it is. It might be confused with an account meant to sign the transaction, as that is also named `account`.

Reply:
do the same in `deploy` function